scripts/clones_analysis/clones_analysis.ipynb

```diff
@@ -0,0 +1,7188 @@
+       "    <span class=\"k\">return</span> <span class=\"n\">binding</span><span class=\"p\">.</span><span class=\"na\">root</span>\n",
+       "<span class=\"p\">}</span>\n",
+       "</pre></div></td>\n",
+       "            </tr>\n",
+       "    </tbody></table>"
+      ],
+      "text/plain": [
+       "<pandas.io.formats.style.Styler at 0x7fb1121e3940>"
+      ]
+     },
+     "execution_count": 102,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pretty_print(unique_clones_medium[[MethodsColumn.METHOD_ID, MethodsColumn.N_UNIQUE_PROJECTS, MethodsColumn.HIGHLIGHTED_CODE]]\n",
+    "             .sort_values(by=[MethodsColumn.N_UNIQUE_PROJECTS], ascending=False)\n",
+    "             .head(n=5))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 105,
+   "id": "fbd52c22",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Index(['project_id', 'method_id', 'file', 'start_line', 'end_line', 'tokens',\n",
+       "       'leader_method_id', 'component_size', 'n_unique_projects_component',\n",
+       "       'n_clones', 'n_inter_clones', 'n_100_clones', 'n_inter_100_clones',\n",
+       "       'n_unique_projects', 'n_unique_100_projects', 'method_text',\n",
+       "       'highlighted_code'],\n",
+       "      dtype='object')"
+      ]
+     },
+     "execution_count": 105,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "unique_clones_medium.columns"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f2ecbe83",
+   "metadata": {},
+   "source": [
+    "**Save exact clones and near-miss clones ratings:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 103,
+   "id": "3a0e26f6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "unique_exact_clones_medium_head = unique_clones_medium.sort_values(by=[MethodsColumn.N_UNIQUE_100_PROJECTS], ascending=False)\\\n",
+    "    .head(n=30)\n",
+    "unique_exact_clones_medium_head.to_csv(f\"{ratings_directory_name}/exact_clones_19_tokens.csv\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 104,
+   "id": "6c276d20",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "unique_clones_medium_head = unique_clones_medium.sort_values(by=[MethodsColumn.N_UNIQUE_PROJECTS], ascending=False).head(n=30)\n",
+    "unique_clones_medium_head.to_csv(f\"{ratings_directory_name}/clones_19_tokens.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2e51cadb",
+   "metadata": {},
+   "source": [
+    "### More clones!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d333ddb",
+   "metadata": {},
+   "source": [
+    "In this part we use clones for methods, that has at least 5 tokens. It is difficult to build a graph for such amount of data, so the analysis part is based on the one pass of adjacency list (there is script for building it)."
```

Comment:
In this part, we use clones for methods that have at least 5 tokens. It is difficult to build a graph for such amount of data, so the analysis part is based on the one pass of adjacency list (there is script for building it).

Reply:
_there is script for building it_

where? :)